Quarkus 3: Using `@TestReactiveTransaction` at the class level doesn't work

### Describe the bug

In Quarkus 3, using `@TestReactiveTransaction` at the class level doesn't work. It causes all the tests in the class to throw an exception.

If the annotation shouldn't be used at the class level then the annotation should be marked with only `@Target({ ElementType.METHOD })` rather than `@Target({ ElementType.TYPE, ElementType.METHOD })`.

Otherwise if it _should_ be able to be used at the class level, then this should be fixed.

### Expected behavior

I would expect that using the annotation on the class would work the same as putting the annotation individually on each test method.

### Actual behavior

If I put the `@TestReactiveTransaction` annotation on a test class, all the tests fail with the following exception. If I move the annotation to each test method then all the tests work fine.

```

java.lang.AssertionError: Reflective call to UniAsserter parameter failed, please file a bug report

	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptor.handleSpecialTestMethod(TestReactiveTransactionalInterceptor.java:79)
	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptor.intercept(TestReactiveTransactionalInterceptor.java:23)
	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptorGenerated_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:38)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:26)
	at io.quarkus.sample.superheroes.hero.repository.HeroRepositoryTests_Subclass.findRandomNotFound(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.test.vertx.RunOnVertxContextTestMethodInvoker$RunTestMethodOnContextHandler.doRun(RunOnVertxContextTestMethodInvoker.java:148)
	at io.quarkus.test.vertx.RunOnVertxContextTestMethodInvoker$RunTestMethodOnContextHandler.handle(RunOnVertxContextTestMethodInvoker.java:137)
	at io.quarkus.test.vertx.RunOnVertxContextTestMethodInvoker$RunTestMethodOnContextHandler.handle(RunOnVertxContextTestMethodInvoker.java:108)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptor.handleSpecialTestMethod(TestReactiveTransactionalInterceptor.java:66)
	... 24 more
Caused by: java.lang.IllegalStateException: Can't get the context safety flag: the current context is not a duplicated context
	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.checkIsSafe(VertxContextSafetyToggle.java:78)
	at io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle.validateContextIfExists(VertxContextSafetyToggle.java:72)
	at io.quarkus.hibernate.reactive.panache.common.runtime.SessionOperations.vertxContext(SessionOperations.java:183)
	at io.quarkus.hibernate.reactive.panache.common.runtime.SessionOperations.withSession(SessionOperations.java:112)
	at io.quarkus.hibernate.reactive.panache.common.runtime.SessionOperations.withTransaction(SessionOperations.java:99)
	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptor$1.apply(TestReactiveTransactionalInterceptor.java:69)
	at io.quarkus.hibernate.reactive.panache.common.runtime.TestReactiveTransactionalInterceptor$1.apply(TestReactiveTransactionalInterceptor.java:66)
	at io.quarkus.test.vertx.DefaultUniAsserter.surroundWith(DefaultUniAsserter.java:129)
	... 29 more
```

### How to Reproduce?

1. Clone the Quarkus superheroes from my fork on the `quarkus3` branch (https://github.com/edeandrea/quarkus-super-heroes/tree/quarkus3)
2. cd into `rest-heroes`
3. Run ` ./mvnw test -Dtest=HeroRepositoryTests`
4. Notice all tests pass
5. Open `src/test/java/io/quarkus/sample/superheroes/hero/repository/HeroRepositoryTests.java`
6. Notice that the `@TestReactiveTransaction` is on all the test methods
7. Remove the `@TestReactiveTransaction` from each test method and place it at the class level
8. Re-run ` ./mvnw test -Dtest=HeroRepositoryTests` and notice that the tests fail.

### Output of `uname -a` or `ver`

`Darwin edeandrea-m1pro 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000 arm64`

### Output of `java -version`

```
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+10 (build 17.0.6+10)
OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (build 17.0.6+10, mixed mode)
```

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

`3.0.0.CR2`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.9.1-bin/320285b4/apache-maven-3.9.1
Java version: 17.0.6, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.6-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "13.3.1", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

--- a/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/TestReactiveTransactionTest.java
+++ b/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/TestReactiveTransactionTest.java
@@ -6,21 +6,17 @@ import org.junit.jupiter.api.Test;
 import io.quarkus.hibernate.reactive.panache.Panache;
 import io.quarkus.test.TestReactiveTransaction;
 import io.quarkus.test.junit.QuarkusTest;
-import io.quarkus.test.vertx.RunOnVertxContext;
 import io.quarkus.test.vertx.UniAsserter;
 
 @QuarkusTest
+@TestReactiveTransaction
 public class TestReactiveTransactionTest {
 
-    @RunOnVertxContext
-    @TestReactiveTransaction
     @Test
     public void testTestTransaction(UniAsserter asserter) {
         asserter.assertNotNull(() -> Panache.currentTransaction());
     }
 
-    @RunOnVertxContext
-    @TestReactiveTransaction
     @BeforeEach
     public void beforeEach(UniAsserter asserter) {
         asserter.assertNotNull(() -> Panache.currentTransaction());

--- a/test-framework/vertx/src/main/java/io/quarkus/test/vertx/RunOnVertxContextTestMethodInvoker.java
+++ b/test-framework/vertx/src/main/java/io/quarkus/test/vertx/RunOnVertxContextTestMethodInvoker.java
@@ -99,7 +99,8 @@ public class RunOnVertxContextTestMethodInvoker implements TestMethodInvoker {
         }
         if (runOnVertxContext == null) {
             // Use duplicated context if @TestReactiveTransaction is present
-            return m.isAnnotationPresent(TestReactiveTransaction.class);
+            return m.isAnnotationPresent(TestReactiveTransaction.class)
+                    || m.getDeclaringClass().isAnnotationPresent(TestReactiveTransaction.class);
         } else {
             return runOnVertxContext.duplicateContext();
         }

